sdk/communication/communication-chat/src/chatClient.ts

```diff
@@ -341,12 +341,32 @@ export class ChatClient {
    */
   public on(event: "participantsRemoved", listener: (e: ParticipantsRemovedEvent) => void): void;
 
-  public on(event: ChatEventId, listener: (e: any) => void): void {
+  /**
+   * Subscribe function for realTimeNotificationConnected.
+   * @param event - The realTimeNotificationConnected Event
+   * @param listener - The listener to handle the event.
+   */
+  public on(event: "realTimeNotificationConnected", listener: () => void): void;
+
+  /**
+   * Subscribe function for realTimeNotificationDisconnected.
+   * @param event - The realTimeNotificationDisconnected Event
+   * @param listener - The listener to handle the event.
+   */
+  public on(event: "realTimeNotificationDisconnected", listener: () => void): void;
+
+  public on(
+    event: ChatEventId | "realTimeNotificationConnected" | "realTimeNotificationDisconnected",
```

Comment:
what type is ChatEventId anyway? It's imported from signalling so I don't think we have good visibility into it in our review tooling. Can we redeclare it locally maybe?

Reply:
ChatEventId is the event names for trouter notification, for example. "chatMessageReceived", "participantsRemoved", etc. Sure I can redeclare it locally